src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -629,7 +638,21 @@ private async Task CalculateTotalAndResourceCounts()
 
         private async Task UpdateParametersAndCompleteJob()
         {
-            (bool success, string error) = await _reindexUtilities.UpdateSearchParameterStatus(_reindexJobRecord.SearchParams.ToList(), _cancellationToken);
+            // here we check if all the resource types which are base types of the search parameter
+            // were reindexed by this job.  If so, then we should mark the search parameters
```

Comment:
>were reindexed by this job [](http://example.com/codeflow?start=15&length=26)

It doesn't seems right.

I reindex one type out of two to make search parameter work for that type.

To fully enable searchparameter I need to run reindex job for both of them, and there is reason why I opt out of running it only for one type.

Which mean user would have to constantly use header based approach to run searches, which would be upsetting.

Reply:
This is true, with this implementation the user would have to continually use the header.  We do not currently track status of whether a searchparameter is fully indexed for a given type, only for all types.  To support searching after indexing only one type, we would need to add more data to the tracking logic and track search parameter status per resource type.